stdlib/importlib/abc.pyi

```diff
@@ -115,8 +107,12 @@ if sys.version_info >= (3, 9):
         def is_file(self) -> bool: ...
         @abstractmethod
         def iterdir(self) -> Iterator[Traversable]: ...
-        @abstractmethod
-        def joinpath(self, child: StrPath) -> Traversable: ...
+        if sys.version_info >= (3, 11):
+            @abstractmethod
+            def joinpath(self, *descendants: str) -> Traversable: ...
+        else:
+            @abstractmethod
+            def joinpath(self, child: str) -> Traversable: ...
```

Comment:
The annotation for this parameter in CPython is `StrPath` -- why should we do differently here? https://github.com/python/cpython/blob/d4d5e605cd74b198a6e21d5bf8fa053de2069aae/Lib/importlib/resources/abc.py#L99

```suggestion
        if sys.version_info >= (3, 11):
            @abstractmethod
            def joinpath(self, *descendants: StrPath) -> Traversable: ...
        else:
            @abstractmethod
            def joinpath(self, child: StrPath) -> Traversable: ...
```

Reply:
The annotation in CPython was copied from the typeshed.  A filesystem path is not valid as an abstract resource.  The whole point of `importlib.resources` is to disassociate package resources from the filesystem so that you might be able to load "files" from e.g. memory, a zip file, over the network, an SQL database, or who knows what.  Accepting `os.PathLike` here makes as much sense as defining module names in terms of filesystem paths.